modules/controlfreec/1.1/controlfreec.smk

```diff
@@ -114,44 +100,120 @@ rule _check_chrFiles:
         touch("references/{genome_build}/freec/chr/.all_done")
 
 
+rule _dbsnp_to_bed:
+    input:
+        vcf = reference_files("genomes/{genome_build}/variation/dbsnp.common_all-151.vcf.gz")
+    output:
+        bed = "references/{genome_build}/freec/dbsnp.common_all-151.bed"
+    shell:
+        op.as_one_line(""" gunzip -c {input.vcf} | awk {{'printf ("%s\\t%s\\t%s\\n", $1,$2-1,$2)'}} | zgrep -v -h "^#" > {output.bed} """)
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _controlfreec_input_bam:
+    input:
+        bam = CFG["inputs"]["sample_bam"],
+        bai = CFG["inputs"]["sample_bai"]
+    output:
+        bam = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{sample_id}.bam",
+        bai = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{sample_id}.bai"
+    run:
+        op.relative_symlink(input.bam, output.bam)
+        op.relative_symlink(input.bai, output.bai)
+
+
+# set-up mpileups for BAF calling
+chroms = list(map(str, range(1, 23))) + ["X", "Y"] 
+chroms = ["chr" + chrom for chrom in chroms]     
+
+
+rule _controlfreec_mpileup_per_chrom:
+    input:
+        bam = CFG["dirs"]["inputs"] + "{seq_type}--{genome_build}/{sample_id}.bam",
+        fastaFile = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        bed = "references/hg38/freec/dbsnp.common_all-151.bed"
+    output: # creates a temporary file for mpileup
+        pileup = temp(CFG["dirs"]["mpileup"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}/{sample_id}.{chrom}.minipileup.pileup.gz"),
+    conda:
+        CFG["conda_envs"]["controlfreec"]
+    threads: 1
+    resources: 
```

Comment:
please add bam=1 here so we can limit number of concurrent jobs running at the same time. Also, in recent modules (like lofreq, linx) we started using resources unpacking - it might be beneficial to use it here too